merlin/models/tf/blocks/core/transformations.py

```diff
@@ -400,3 +401,23 @@ def call(self, inputs, training=False, **kwargs) -> tf.Tensor:
         logits = tf.nn.bias_add(logits, self.bias)
 
         return logits
+
+
+@Block.registry.register_with_multiple_names("label_to_onehot")
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class LabelToOneHot(Block):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
```

Comment:
Doesn’t the shape change here?

Reply:
I debugged it and input_shape comes with the correct dims `TensorShape(batch size, # items)`. In this transform, we only apply one-hot encoding to the `labels`, which in fact have there shape changed to be equal to `predictions`.